cdap-docs/developers-manual/source/data-exploration/object-mapped-tables.rst

```diff
@@ -0,0 +1,118 @@
+If you wish to name the key column differently, because your object contains a field named "key" for example, you 
+can do so by providing an additional property when creating your Dataset. You can also set the key type to
+a string if desired::
+  
+  @Override
+  public void configure() {
+    try {
+      createDataset("purchases", ObjectMappedTable.class,
+                    ObjectMappedTableProperties.builder()
+                      .setType(Purchase.class)
+                      .setExploreKeyName("rowkey")
+                      .setExploreKeyType(Schema.Type.STRING)
+                      .build()
+                   );
+    } catch (UnsupportedTypeException e) {
+      // This exception is thrown by ObjectMappedTable if its parameter type cannot be
+      // (de)serialized (for example, if it is an interface and not a class, then there is
+      // no auto-magic way deserialize an object.) In this case that will not happen
+      // because Purchase is an actual classes.
+      throw new RuntimeException(e);
+    }
+  } 
+
+Creating the Dataset in this manner would result in a different column name and type for the object key:: 
+
+  (rowkey STRING, catalogid STRING, customer STRING, price INT, product STRING, purchasetime BIGINT, quantity INT))
+
+.. _sql-limitations:
+
+Limitations
+-----------
+* The record type must be a structured type, that is, a Java class with fields. This is because SQL tables require
+  a structure type at the top level. The fields must also be primitives. That is, they must be an int, Integer,
+  float, Float, double, Double, bool, Boolean, String, byte[], or ByteBuffer. In addition, UUID is supported and
+  will translate into a binary field.
+
+* The record type must be that of an actual Java class, not an interface. The reason is that interfaces only define
+  methods but not fields; hence, reflection would not be able to derive any fields or types from the interface.
+
+* Fields of a class that are declared static or transient are ignored during schema generation. This means that the
+  record type must have at least one non-transient and non-static field. For example,
+  the ``java.util.Date`` class has only static and transient fields. Therefore a record type of ``Date`` is not
+  supported and will result in an exception when the Dataset is created.
+
+* You cannot insert data into an ObjectMappedTable using SQL.
+
+Formulating Queries
+-------------------
+When creating your queries, keep these limitations in mind:
+
+- The query syntax of CDAP is a subset of the variant of SQL that was first defined by Apache Hive.
+- The SQL commands ``UPDATE`` and ``DELETE`` are not allowed on CDAP Datasets.
+- When addressing your datasets in queries, you need to prefix the data set name with the CDAP
+  namespace ``cdap_user_``. For example, if your Dataset is named ``Purchases``, then the corresponding table
+  name is ``cdap_user_purchases``. Note that the table name is lower-case.
+
```

Comment:
Is there any other namespace issues? I'm a little concerned about saying "CDAP namespace"; wouldn't that confuse people versus the "other" CDAP namespace(s)? Are these tables namespaced otherwise? If you create a table in the default namespace and then an identical table in a different namespace (say, "dev"), what distinguishes them at the hive level? Don't you need to add the namespace in there somewhere?